google-cloud-storage/src/main/java/com/google/cloud/storage/UnifiedOpts.java

```diff
@@ -661,13 +669,209 @@ public Mapper<ListObjectsRequest.Builder> listObjects() {
+      return b -> {
+        String collect =
+            val.stream().map(NamedField::getApiaryName).collect(Collectors.joining(","));
+        return b.put(StorageRpc.Option.FIELDS, collect);
+      };
+    }
+
+    @Override
+    public Mapper<GetBucketRequest.Builder> getBucket() {
+      return b -> b.setReadMask(FieldMask.newBuilder().addAllPaths(getPaths()).build());
+    }
+
+    @Override
+    public Mapper<ListBucketsRequest.Builder> listBuckets() {
+      return b -> b.setReadMask(FieldMask.newBuilder().addAllPaths(getPaths()).build());
+    }
+
+    @Override
+    public Mapper<GetObjectRequest.Builder> getObject() {
+      return b -> b.setReadMask(FieldMask.newBuilder().addAllPaths(getPaths()).build());
+    }
+
+    @Override
+    public Mapper<ListObjectsRequest.Builder> listObjects() {
+      return b -> b.setReadMask(FieldMask.newBuilder().addAllPaths(getPaths()).build());
+    }
+
+    @Override
+    public Mapper<ReadObjectRequest.Builder> readObject() {
+      return b -> b.setReadMask(FieldMask.newBuilder().addAllPaths(getPaths()).build());
+    }
+
+    /**
+     * Define a decoder which can clear out any fields which may have not been selected.
+     *
+     * <p>This approach, isn't ideal at the backside after decoding has already taken place.
+     * However, refactoring the whole model pipeline for both json and grpc is going to be a large
+     * change.
+     */
+    Decoder<Blob, Blob> clearBlobFields() {
+      return b -> {
+        if (val.isEmpty()) {
+          return b;
+        } else {
+          Set<String> names = getPaths();
+          Blob.Builder bldr = b.toBuilder();
+          blobInfoFieldClearers.entrySet().stream()
+              .filter(e -> !names.contains(e.getKey()))
+              .map(Entry::getValue)
+              .forEach(m -> m.apply(bldr));
+          return bldr.build();
+        }
+      };
+    }
+
+    /**
+     * Define a decoder which can clear out any fields which may have not been selected.
+     *
+     * <p>This approach, isn't ideal at the backside after decoding has already taken place.
+     * However, refactoring the whole model pipeline for both json and grpc is going to be a large
+     * change.
+     */
+    Decoder<Bucket, Bucket> clearBucketFields() {
+      return b -> {
+        if (val.isEmpty()) {
+          return b;
+        } else {
+          Set<String> names = getPaths();
+          Bucket.Builder bldr = b.toBuilder();
+          bucketInfoFieldClearers.entrySet().stream()
+              .filter(e -> !names.contains(e.getKey()))
+              .map(Entry::getValue)
+              .forEach(m -> m.apply(bldr));
+          return bldr.build();
+        }
+      };
+    }
+
+    private Set<String> getPaths() {
+      //noinspection Guava
+      return val.stream()
+          .map(NamedField::stripPrefix)
+          .map(NamedField::getGrpcName)
+          .filter(not(grpcExcludedFields::contains))
```

Comment:
nit: probably want to filter and then get the GRPC name because we shouldn't have GRPC names for non GRPC fields?

Reply:
The exclusions are a `ImmutableSet<String>` because some fields are hard coded as strings for apiary ("nextPageToken") and don't have defined values we can reference like we can for `Storage.BlobField` and `Storage.BucketField`.